src/libraries/System.Data.Common/src/System/Data/Common/DbConnectionStringBuilder.cs

```diff
@@ -562,16 +565,22 @@ private PropertyDescriptorCollection GetProperties(Attribute[]? attributes)
             return new PropertyDescriptorCollection(filteredPropertiesArray);
         }
 
-// TODO: Enable after System.ComponentModel.TypeConverter is annotated
+        // TODO-NULLABLE: Enable after System.ComponentModel.TypeConverter is annotated
 #nullable disable
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "The type of component is statically known. This class is marked with [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]")]
```

Comment:
@vitek-karas - is this going to work if we have `[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]` here, but these methods don't call `this.GetType()`?

Reply:
I think `GetAttributes()` is the only "iffy" one here. `ClassName` and `ComponentName` are pretty straight forward they can be suppressed.

Reply:
Correct - this won't work as-is. The method would have to call `this.GetType()`.
Is it possible to rewrite this to use the overloads on `TypeDescriptor` which take the `Type` parameter? That would make this correct without suppressions.

Reply:
As for `GetAttributes` - so far the attributes in linker where handled:
* Linker doesn't remove attributes
* Unless it's told to do so explicitly - which we do quite a bit in Blazor, but typically we don't (there are few exceptions, like nullability attributes)

I would probably not worry about attributes - if this becomes a problem it's a larger issue not confined to S.D.Common.

Reply:
I don't think that will work - when we query using object and object is implementing IComponent it merges it's own data with the type data, we'll lose that info if we go directly with type

Reply:
In that case the safest way (which is still pretty hacky unfortunately) would be to call `this.GetType()` (and don't use the result 😞 ). This will basically tell the linker that it should mark the derived type appropriately, but validation won't happen, so you would need to suppress still.